boards/beagle/beaglebone_ai64/beaglebone_ai64_j721e_main_r5f0_0_defconfig

```diff
@@ -0,0 +1,18 @@
+# Copyright (C) 2023 BeagleBoard.org Foundation
+# Copyright (C) 2023 S Prashanth
+#
+# SPDX-License-Identifier: Apache-2.0
+
+# Zephyr Kernel Configuration
+CONFIG_XIP=n
+
+# Enable Pinctrl
+CONFIG_PINCTRL=y
```

Comment:
Not allowed

Reply:
@glneo You need to select pinctrl at `soc/ti/k3/am6x/Kconfig`.
See https://github.com/zephyrproject-rtos/zephyr/pull/64718#discussion_r1734288793

Reply:
Seems PINCTRL was already set to default y in `soc/ti/k3/am6x/Kconfig.defconfig` so this line wasn't needed anyway, removed.

Reply:
> Seems PINCTRL was already set to default y in soc/ti/k3/am6x/Kconfig.defconfig so this line wasn't needed anyway, removed.

Even worse, that needs to be removed and it done properly

Reply:
@nordicjm It should be removed, but should that really be part of this PR? I think it might be sufficient to just add `select PINCTRL` in `SOC_SERIES_AM6X_R5` and leave cleaning up am6x in a separate PR.

Reply:
It's wrong and needs to be done properly

Reply:
> It's wrong and needs to be done properly

It became the wrong way to do it at some point after am6x support landed.

This commit only adds AM6X_R5 support, and thus, yes, it should be done properly for anything new. As such `select PINCTRL` should be added in `SOC_SERIES_AM6X_R5`.

But I don't see why fixing all the other AM6x boards should be part of this PR.

Reply:
This is adding a new SoC, it needs to add the selections properly, right now the selections are wrong. Will doing it properly for this SoC fix the other SoC? Yes

Reply:
> This is adding a new SoC, it needs to add the selections properly, right now the selections are wrong. Will doing it properly for this SoC fix the other SoC? Yes

Ok, let me clarify something. For me, doing it properly would mean 
1. Removing `CONFIG_PINCTRL=y` selection from `soc/ti/k3/am6x/Kconfig.defconfig`.
2. Adding `select PINCTRL` to `SOC_SERIES_AM6X_A53`, `SOC_SERIES_AM6X_M4` 
3. Removing `CONFIG_PINCTRL=y` from any board using these SOCs

Are you referring to only doing 1 and 2 in the PR? Or should 3 also be handled here? Because this PR actually only adds `AM6X_R5` and does not touch `AM6X_A53` and `AM6X_M4`.

Reply:
All 3. Honestly the time with all these messages is more than the total time it would take to fix it and update the PR

Reply:
Also, it might be good to link this issue here for more context on what @nordicjm is referring to: https://github.com/zephyrproject-rtos/zephyr/issues/78619

Reply:
> All 3. Honestly the time with all these messages is more than the total time it would take to fix it and update the PR

I mean sure, if someone already knew what you wanted them to do. The thing is that the number of boards that correctly do it is in the minority right now, so it would be great if someone linked https://github.com/zephyrproject-rtos/zephyr/issues/78619 . I only found out that issue existed today.

In my Beagleplay PR, the only board that followed the pattern was esp32, and that was basically it. So I am trying to solidly define what needs to be done to get this merged.

Reply:
There's over 600 boards in zephyr, that PR links the boards that do it wrong in the first post and the SoCs that do it wrong in the bottom post (which I'm thankful for you for mentioning earlier as I was not aware SoCs were doing it wrong also), there's about 150 or so checkboxes in that issue, so that's a minority of the 600+ supported boards. The way to do it for board and SoC is the same, SoC drivers select it, if a board needs it because it has a custom .c file (this is the only valid reason a board should select it) then the board Kconfig file (note: `Kconfig` not `Kconfig.<board>`) selects it in the same way

Reply:
Okay, will add a patch fixing the PINCTRL selection for other boards and SoCs as part of this series too.